SolaceProducts/solace-apache-beam
# Fix and Cleanup Tests

**Test Changes:**
* Fix test credential acquisition to be more flexible.
* Use testcontainers when testing against the direct runner.
* Use solace-semp-v2-client as SEMP client for tests.
* Remove `it` profile for running integration tests. These should always run by default. If a dev needs to skip them, use `-DskipITs`.
* Refactor sample tests into JUnit.

**Github Action Workflow Changes:**
* Separate workflow for Dataflow testing to fix PR action failures from forks.
* Add Maven caching.
* Add job to skip duplicate workflow run.
* Add `workflow_dispatch` trigger.
* Upgrade `setup-java` action to `v2`.
